slither/utils/code_generation.py

```diff
@@ -56,34 +116,63 @@ def generate_interface_function_signature(func: "FunctionContract") -> Optional[
         Returns None if the function is private or internal, or is a constructor/fallback/receive.
     """
 
-    name, parameters, return_vars = func.signature
+    name, _, _ = func.signature
     if (
         func not in func.contract.functions_entry_points
         or func.is_constructor
         or func.is_fallback
         or func.is_receive
     ):
         return None
-    view = " view" if func.view else ""
+    view = " view" if func.view and not func.pure else ""
     pure = " pure" if func.pure else ""
     payable = " payable" if func.payable else ""
     returns = [
         convert_type_for_solidity_signature_to_string(ret.type).replace("(", "").replace(")", "")
+        if unroll_structs
+        else convert_type_for_solidity_signature_to_string(ret.type)
+        .replace("(", "")
+        .replace(")", "")
+        + f" {ret.location}"
+        if isinstance(ret.type, ArrayType)
+        and isinstance(ret.type.type, (UserDefinedType, ElementaryType))
+        else f"{str(ret.type.type)} memory"
+        if isinstance(ret.type, UserDefinedType) and isinstance(ret.type.type, (Structure, Enum))
+        else "address"
+        if isinstance(ret.type, UserDefinedType) and isinstance(ret.type.type, Contract)
+        else f"{ret.type} {ret.location}"
+        if ret.type.is_dynamic
+        else str(ret.type)
         for ret in func.returns
     ]
     parameters = [
         convert_type_for_solidity_signature_to_string(param.type).replace("(", "").replace(")", "")
+        if unroll_structs
+        else convert_type_for_solidity_signature_to_string(param.type)
+        .replace("(", "")
+        .replace(")", "")
+        + f" {param.location}"
+        if isinstance(param.type, ArrayType)
+        and isinstance(param.type.type, (UserDefinedType, ElementaryType))
+        else f"{str(param.type.type)} memory"
+        if isinstance(param.type, UserDefinedType)
+        and isinstance(param.type.type, (Structure, Enum))
+        else "address"
+        if isinstance(param.type, UserDefinedType) and isinstance(param.type.type, Contract)
+        else f"{param.type} {param.location}"
+        if param.type.is_dynamic
+        else str(param.type)
```

Comment:
Could you extract this into a helper function please? 
Like 
```
def generate_interface_function_signature
     def helper
```

Reply:
Done!